### PR TITLE
cli: Support CXX and CXXFLAGS

### PR DESCRIPTION
**Pull request by jayvdb**
Adds plumbing for using with g++ only, testing with astyle as
an alternative formatter.

Also allow CLANG_FORMAT_STYLE to determine the clang-format
style to be used.

### CODE REVIEW

File: py2many/cli.py [hunk 1]
```diff
@@ -252,19 +275,27 @@ def _process_once(settings, filename, outdir):
...
-            print(f"Error: Could not reformat: {output_path}")
-            return False
-    if settings.ext == ".kt":
-        # ktlint formatter needs to be invoked twice before output is lint free
-        if subprocess.call([*settings.formatter, output_path]):
-            print(f"Error: Could not reformat: {output_path}")
+        cmd = _create_cmd(settings.formatter, output_path)
+        proc = run(cmd, env=env)
+        # format.jl exit code is unreliable
+        if settings.ext == ".jl":
```
Comment (jayvdb) [inline py2many/cli.py:283]: this should be moved inside the `if proc.returncode:` below after the `print`

File: tests/test_cli.py [hunk 2]
```diff
@@ -192,6 +216,69 @@ def test_cli(self, case, lang):
...
+        if not spawn.find_executable(env["CXX"]):
+            raise unittest.SkipTest(f"{env['CXX']} not available")
+
+        settings = _get_all_settings(Mock(indent=4), env=env)[lang]
+        assert settings.linter[0].startswith("g++")
+
+        if not spawn.find_executable("astyle"):
+            raise unittest.SkipTest("astyle not available")
+
+        settings.formatter = ["astyle"]
```
Comment (adsharma) [inline tests/test_cli.py:241]: Looks like astyle is an alternative formatter and has nothing to do with whether CXX=gcc or clang.

**Review by adsharma — APPROVED** [review]
Good to know that generated code works with g++ as well!

Let me know if you want to rebase it yourself.

File: tests/test_cli.py (hunk 2 above)
Comment (jayvdb) [reply tests/test_cli.py:241]: I'd like to remove the dependency on llvm, and clang-format is part of that goal. c.f. https://github.com/adsharma/py2many/issues/21  I guess adding a `--formatter=...` and `--linter=...` cli args is how we would expose that.  For now using astyle here ensures this group of tests doesnt use llvm.

**Pull request merged**